zio-http/src/main/scala/zhttp/service/server/ServerSSLBuilder.scala

```diff
@@ -0,0 +1,82 @@
+package zhttp.service.server
+
+import io.netty.handler.codec.http2.Http2SecurityUtil
+import io.netty.handler.ssl.ApplicationProtocolConfig.{
+  Protocol,
+  SelectedListenerFailureBehavior,
+  SelectorFailureBehavior,
+}
+import io.netty.handler.ssl._
+
+import java.io.InputStream
+import java.security.KeyStore
+import javax.net.ssl.KeyManagerFactory
+
+object ServerSSLBuilder {
+
+  case class ServerSSLOptions(
+    sslContextBuilder: SslContextBuilder,
+    httpBehaviour: SSLHttpBehaviour = SSLHttpBehaviour.Redirect,
+  )
+
+  sealed trait SSLHttpBehaviour
+
+  object SSLHttpBehaviour {
+
+    case object Redirect extends SSLHttpBehaviour
+
+    case object Accept extends SSLHttpBehaviour
+
+    case object Fail extends SSLHttpBehaviour
+
+  }
+
+  def ctxFromKeystore(
+    keyStoreInputStream: InputStream,
+    keyStorePassword: String,
+    certPassword: String,
+  ): SslContextBuilder = {
+    val keyStore: KeyStore = KeyStore.getInstance("JKS")
+    keyStore.load(keyStoreInputStream, keyStorePassword.toCharArray)
+    val kmf                = KeyManagerFactory.getInstance("SunX509")
+    kmf.init(keyStore, certPassword.toCharArray)
+    SslContextBuilder
+      .forServer(kmf)
+      .sslProvider(SslProvider.JDK)
+  }
+
+  def ctxFromCert(certInputStream: InputStream, keyInputStream: InputStream): SslContextBuilder = {
+    SslContextBuilder
+      .forServer(certInputStream, keyInputStream)
+      .sslProvider(SslProvider.JDK)
+  }
+
+  def build(sslContextBuilder: SslContextBuilder, enableHttp2: Boolean = false): SslContext = {
+    if (enableHttp2) {
+      sslContextBuilder
+        .ciphers(Http2SecurityUtil.CIPHERS, SupportedCipherSuiteFilter.INSTANCE)
```

Comment:
I think we should use `builder pattern` to build the arguments (eg. ciphers, ApplicationProtocolConfig) for`SSLContextBuilder` and that should have `build` method which actually calls `sslContextBuilder`